source/threads.tex

```diff
@@ -306,6 +330,109 @@
 \tcode{true} if the lock was acquired, otherwise \tcode{false}.
 \end{itemdescr}
 
+\rSec3[thread.req.lockable.shared]{\oldconcept{SharedLockable} requirements}
+
+\pnum
+A type \tcode{L} meets the \oldconcept{SharedLockable} requirements if
+the following expressions are well-formed, have the specified semantics, and
+the expression \tcode{m.try_lock_shared()} has type \tcode{bool}
+(\tcode{m} denotes a value of type \tcode{L}):
+
+\begin{itemdecl}
+m.lock_shared()
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Blocks until a lock can be acquired for the current execution agent.
+If an exception is thrown then a lock shall not have been acquired for
+the current execution agent.
+\end{itemdescr}
+
+\begin{itemdecl}
+m.try_lock_shared()
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Attempts to acquire a lock for the current execution agent without blocking.
+If an exception is thrown then a lock shall not have been acquired for
+the current execution agent.
+
+\pnum
+\returns
+\tcode{true} if the lock was acquired, \tcode{false} otherwise.
+\end{itemdescr}
+
+\begin{itemdecl}
+m.unlock_shared()
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+The current execution agent holds a shared lock on \tcode{m}.
+
+\pnum
+\effects
+Releases a shared lock on \tcode{m} held by the current execution agent.
+
+\pnum
+\throws
+Nothing.
+\end{itemdescr}
+
+\rSec3[thread.req.lockable.shared.timed]{\oldconcept{SharedTimedLockable} requirements}
+
+\pnum
+A type \tcode{L} meets the \oldconcept{SharedTimedLockable} requirements if
+it meets the \oldconcept{SharedLockable} requirements, and
+the following expressions are well-formed, have type \tcode{bool}, and
+have the specified semantics
+(\tcode{m} denotes a value of type \tcode{L},
+\tcode{rel_-time} denotes a value of a specialization of \tcode{chrono::duration}, and
+\tcode{abs_-time} denotes a value of a specialization of \tcode{chrono::time_-point}).
+
+\begin{itemdecl}
+m.try_lock_shared_for(rel_time)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Attempts to acquire a lock for the current execution agent within
+the relative timeout\iref{thread.req.timing} specified by \tcode{rel_-time}.
+The function will not return within the timeout specified by \tcode{rel_-time}
+unless it has obtained a lock on \tcode{m} for the current execution agent.
+If an exception is thrown then a lock has not been acquired for
+the current execution agent.
+
+\pnum
+\returns
+\tcode{true} if the lock was acquired, \tcode{false} otherwise.
+\end{itemdescr}
+
+\begin{itemdecl}
+m.try_lock_shared_until(abs_time)
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Attempts to acquire a lock for the current execution agent before
+the absolute timeout\iref{thread.req.timing} specified by \tcode{abs_-time}.
+The function will not return before the timeout specified by \tcode{abs_-time}
```

Comment:
What's this: "abs_-time". That looks broken. Same for rel_-time above.

Reply:
I guess the paper rendered strangely with those non-ascii chars. Fixed.